src/components/Personalization/createComponent.js

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2020 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import { noop } from "../../utils";
+
+export default ({
+  config,
+  logger,
+  onResponseHandler,
+  onClickHandler,
+  hideContainers,
+  showContainers,
+  hasScopes,
+  isAuthoringModeEnabled,
+  getDecisionScopes,
+  mergeQuery,
+  createQueryDetails
+}) => {
+  const { prehidingStyle } = config;
+
+  return {
+    lifecycle: {
+      onBeforeEvent({
+        event,
+        renderDecisions,
+        decisionScopes = [],
+        onResponse = noop,
+        onRequestFailure = noop
+      }) {
+        if (isAuthoringModeEnabled()) {
+          logger.warn("Rendering is disabled, authoring mode.");
+
+          // If we are in authoring mode we disable personalization
+          mergeQuery(event, { enabled: false });
+          return;
+        }
+
+        const scopes = getDecisionScopes(renderDecisions, decisionScopes);
+
+        if (!hasScopes(scopes)) {
+          logger.info("Personalization is skipped.");
```

Comment:
I wonder if this is necessary, or if we need more info, like a reason why personalization is skipped. thought?

Reply:
because `renderDecisions` is false and `decisionScopes` is empty, or missing?

Reply:
Personalization is skipped because renderDecisions is false and decisionScopes is empty.
like this?

Reply:
@ninaceban I'm wondering if this message is providing enough value to keep it it around. Is the concern that the user might not know why personalization content isn't being rendered?